mslice/cli/__init__.py

```diff
@@ -1,2 +1,93 @@
+    if not workspace_exists(workspace):
+        raise TypeError('InputWorkspace %s could not be found.' % workspace)
+
+
+def _check_workspace_type(workspace, correct_type):
+    """Check a PSD workspace is MatrixWorkspace, or non-PSD is the specified type"""
+    if workspace.is_PSD:
+        if isinstance(workspace, MatrixWorkspace):
+            raise RuntimeError("Incorrect workspace type - run MakeProjection first.")
+        if not isinstance(workspace, correct_type):
+            raise RuntimeError("Incorrect workspace type.")
+    else:
+        if not isinstance(workspace, MatrixWorkspace):
+            raise RuntimeError("Incorrect workspace type.")
+
+
+def PlotSlice(InputWorkspace, IntensityStart="", IntensityEnd="", Colormap=DEFAULT_CMAP):
+    """
+    Creates mslice standard matplotlib plot of a slice workspace.
+
+    :param InputWorkspace: Workspace to plot. The parameter can be either a python
+    handle to the workspace OR the workspace name as a string.
+    :param IntensityStart: Lower bound of the intensity axis (colorbar)
+    :param IntensityEnd: Upper bound of the intensity axis (colorbar)
+    :param Colormap: Colormap name as a string. Default is 'viridis'.
+    :return:
+    """
+    _check_workspace_name(InputWorkspace)
+    workspace = get_workspace_handle(InputWorkspace)
+    _check_workspace_type(workspace, HistogramWorkspace)
+    slice_presenter = app.MAIN_WINDOW.slice_plotter_presenter
+    slice_presenter.change_intensity(workspace.name, IntensityStart, IntensityEnd)
+    slice_presenter.change_colourmap(workspace.name, Colormap)
+    slice_presenter.plot_from_cache(workspace)
+
+
+def PlotCut(InputWorkspace, IntensityStart=0, IntensityEnd=0, PlotOver=False):
+    """
+    Create mslice standard matplotlib plot of a cut workspace.
+
+    :param InputWorkspace: Workspace to cut. The parameter can be either a python handle to the workspace
+    OR the workspace name as a string.
+    :param IntensityStart: Lower bound of the y axis
+    :param IntensityEnd: Upper bound of the y axis
+    :param PlotOver: if true the cut will be plotted on an existing figure.
+    :return:
+    """
+    _check_workspace_name(InputWorkspace)
+    workspace = get_workspace_handle(InputWorkspace)
+    if not isinstance(workspace, HistogramWorkspace):
+        raise RuntimeError("Incorrect workspace type.")
+    if IntensityStart == 0 and IntensityEnd == 0:
+        intensity_range = None
+    else:
+        intensity_range = (IntensityStart, IntensityEnd)
+    app.MAIN_WINDOW.cut_plotter_presenter.plot_cut_from_workspace(workspace, intensity_range=intensity_range,
+                                                                  plot_over=PlotOver)
+
+
+class MSliceAxes(Axes):
+    name = 'mslice'
+
+    def plot(self, *args, **kwargs):
+
+        if validate_args(*args):
+            mantid.kernel.logger.debug('using mantid.plots.plotfunctions')
+            if is_cut(*args):
```

Comment:
This nearly there except I think we want to keep the plot types separate. `plot` should just create a `cut` and a separate command `pcolormesh` (and its variants) will create the slice.